model/src/main/kotlin/com/testerum/model/resources/http/response/xml_body/XmlRootJsObject.kt

```diff
@@ -0,0 +1,94 @@
+package com.testerum.model.resources.http.response.xml_body
+
+import jdk.nashorn.api.scripting.AbstractJSObject
+import org.jdom2.Attribute
+import org.jdom2.Content
+import org.jdom2.Document
+import org.jdom2.Element
+import org.jdom2.Text
+import org.jdom2.input.SAXBuilder
+import org.jdom2.xpath.XPathFactory
+
+const val XPATH_FUNCTION_NAME = "xpath"
+
+class XmlRootJsObject(private val xmlAsUtf8String: String) : AbstractJSObject() {
+    var rootElement: Element? = null
+    var document: Document? = null
+
+    private fun initXmlTree() {
+        if (rootElement != null) {
+            return
+        }
+
+        document = SAXBuilder().build(xmlAsUtf8String.byteInputStream())
+        rootElement = document!!.rootElement
+    }
+
+    override fun hasMember(name: String): Boolean {
+        initXmlTree()
+
+        return rootElement?.name == name
+    }
+
+    override fun getMember(name: String): Any? {
+        initXmlTree()
+
+        when (name) {
+            XPATH_FUNCTION_NAME -> return EvalXPathFunction(document!!)
+        }
+
+        if (rootElement?.name != name) {
+            throw RuntimeException("""The XML node "$name" can not be found!""")
+        }
```

Comment:
I don't think this is needed. ``XmlNodeJsObject`` will try to get the child by that name, right?